### ISSUE TITLE
DOC: RTD for v5.2.x and v5.0.x failing with sphinx_gallery error

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#14201]
Might have to pin sphinx-gallery, not sure. Don't see this on `main`.

Example log: https://readthedocs.org/projects/astropy/builds/18961672/

```
build succeeded.

The HTML pages are in _build/html.
Updating searchtools for Read the Docs search... 
Sphinx-Gallery successfully executed 9 out of 9 files subselected by:

    gallery_conf["filename_pattern"] = '^((?!skip_).)*$'
    gallery_conf["ignore_pattern"]   = '__init__\\.py'

after excluding 0 files that had previously been run (based on MD5).

embedding documentation hyperlinks...

Traceback (most recent call last):
  File ".../sphinx/events.py", line 111, in emit
    results.append(listener.handler(self.app, *args))
  File ".../sphinx_gallery/docs_resolv.py", line 470, in embed_code_links
    _embed_code_links(app, gallery_conf, gallery_dir)
  File ".../sphinx_gallery/docs_resolv.py", line 327, in _embed_code_links
    doc_resolvers[this_module] = SphinxDocLinkResolver(
  File ".../sphinx_gallery/docs_resolv.py", line 186, in __init__
    self._searchindex = js_index.loads(sindex)
  File ".../sphinx/search/__init__.py", line 179, in loads
    return jsdump.loads(data)
  File ".../sphinx/util/jsdump.py", line 184, in loads
    raise ValueError("read error at pos %d" % i)
ValueError: read error at pos 12

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File ".../sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File ".../sphinx/application.py", line 389, in build
    self.events.emit('build-finished', None)
  File ".../sphinx/events.py", line 119, in emit
    raise ExtensionError(__("Handler %r for event %r threw an exception") %
sphinx.errors.ExtensionError: Handler <function embed_code_links at 0x7f596ac15ca0> for event 'build-finished' threw an exception (exception: read error at pos 12)

Extension error (sphinx_gallery.docs_resolv):
Handler <function embed_code_links at 0x7f596ac15ca0> for event 'build-finished' threw an exception (exception: read error at pos 12)
```

**Comment by saimn** [comment]
Probably same as https://github.com/astropy/astropy/commit/84fb7f08e527fb2cbe99c5c81e0a04f29740b9ef (which has not been backported)

**Comment by pllim** [comment]
Ohhhh... no wonder it looks familiar. 😆 

Last successful log: https://readthedocs.org/api/v2/build/18872323.txt

**Comment by pllim** [comment]
A simple backport of #14204 did not fix this error for v5.2.x branch. I don't know why. I think I am just going to disable sphinx gallery in both v5.2.x and v5.0.x . People can still look at gallery in older releases and in latest, and there are better stuff over at tutorials.

Most of the doc is better than no doc at all.

**Comment by pllim** [comment]
Turns out I needed both of this in the backports:

* https://github.com/astropy/astropy/pull/14140
* https://github.com/astropy/astropy/pull/14171

The actual backports:

* https://github.com/astropy/astropy/pull/14206
* https://github.com/astropy/astropy/pull/14224
